src/layout/monitor.rs

```diff
@@ -890,7 +890,17 @@ impl<W: LayoutElement> Monitor<W> {
         }
 
         let previous_workspace_id = self.previous_workspace_id;
-        self.activate_workspace(new_idx);
+
+        // Only refocus the workspace if it was already focused
+        if self.active_workspace_idx == old_idx {
+            self.activate_workspace(new_idx);
+        // If the workspace order was switched so that the current workspace moved down the workspace stack, focus correctly
+        } else if new_idx <= self.active_workspace_idx && old_idx > self.active_workspace_idx {
+            self.activate_workspace(self.active_workspace_idx + 1);
```

Comment:
I think this should just set self.active_workspace_idx directly. Because updating previous_workspace_id as done in activate_workspace() doesn't make sense in these two cases.